src/main/java/de/tum/in/test/api/structural/testutils/ClassNameScanner.java

```diff
@@ -318,4 +336,42 @@ private void walkProjectFileStructure(String assignmentFolderName, File node,
 			}
 		}
 	}
+
+	static boolean isMisspelledWithHighProbability(String a, String b) {
+		/**
+		 * This based on observations and experiments with
+		 * https://github.com/src-d/datasets/tree/master/Typos and collections real (not
+		 * misspelled) classes.
+		 */
+		/*
+		 * This is a fast check which should work often and not be a problem for the
+		 * user to spot (as this requires a significant length difference). Such "long"
+		 * typos seem to occur almost never by accident.
+		 */
+		int lengthDifferenceAbs = Math.abs(a.length() - b.length());
+		if (lengthDifferenceAbs > 2)
+			return false;
+		/**
+		 * This is the case for most typos, simply one missing or added character or two
+		 * next to each other are swapped. (We only use this rule for strings with
+		 * length of at least two)
+		 */
+		double distance = DAMERAU_LEVENSHTEIN.distance(a, b);
+		if (distance <= 1.0 && Math.max(a.length(), b.length()) > 2)
+			return true;
+		/**
+		 * We accept everything with a distance below three as typo. At three and above,
+		 * misspelled identifiers can be easily recognized by a human or might not be
+		 * spelling errors.
+		 */
+		if (distance > 2)
```

Comment:
Is the comment false or is the distance literal incorrect? Souldn't it be "with a distance below **two** as typo"